jax/core.py

```diff
@@ -371,11 +371,18 @@ class Tracer(object):
   __slots__ = ['_trace', '__weakref__']
 
   def __array__(self, *args, **kw):
-    raise Exception("Tracer can't be used with raw numpy functions. "
-                    "You might have\n"
-                    "  import numpy as np\n"
-                    "instead of\n"
-                    "  import jax.numpy as jnp")
+    msg = ("The numpy.ndarray conversion method __array__() was called on "
+           f"the JAX Tracer object {self}.\n\n"
+           "This error can occurr when a JAX Tracer object is passed to a raw "
```

Comment:
```suggestion
           "This error can occur when a JAX Tracer object is passed to a raw "
```

Reply:
I have never spelled that word correctly on my first try.